File: college DSA/searchbst.py
class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

def search_bst(root, target):
    if root is None:
        return False
    if root.value == target:
        return True
    if target < root.value:
        return search_bst(root.left, target)
    else:
        return search_bst(root.right, target)

# Example usage:
root = Node(5)
root.left = Node(2)
root.right = Node(8)
root.left.left = Node(1)
root.left.right = Node(3)
root.right.left = Node(6)
root.right.right = Node(9)

target = 3
if search_bst(root, target):
    print(f"Found {target} in the BST!")
else:
    print(f"{target} not found in the BST!")

File: college DSA/searchbst.java
public class searchbst {
    public static class Node{  
        int data;  
        Node left;  
        Node right;  
  
        public Node(int data){  
            this.data = data;  
            this.left = null;  
            this.right = null;  
        }  
      }   
      public Node root;  
  
      public searchbst(){  
          root = null;  
      }  
      public void insert(int data) {    
          Node newNode = new Node(data);   
          if(root == null){  
              root = newNode;  
              return;  
            }  
          else {   
              Node current = root, parent = null;  
  
              while(root!=null) {    
                  parent = current;  
                  if(data < current.data) {  
                      current = current.left;  
                      if(current == null) {  
                          parent.left = newNode;  
                          return;  
                      }  
                  }  
                  else {  
                      current = current.right;  
                      if(current == null) {  
                          parent.right = newNode;  
                          return;  
                      }  
                  }  
              }  
          }  
      }  
   
      public Node minNode(Node root) {  
          if (root.left != null)  
              return minNode(root.left);  
          else  
              return root;  
      }    
    //   public void inorderTraversal(Node node) {  
    //       if(root == null){  
    //           System.out.println("Tree is empty");  
    //           return;  
    //        }  
    //       else {  
  
    //           if(node.left!= null)  
    //               inorderTraversal(node.left);  
    //           System.out.print(node.data );  
    //           if(node.right!= null)  
    //               inorderTraversal(node.right);  
  
    //       }  
    //   } 
      public static void main(String[] args) {  
  
        BST bt = new BST();   
        bt.insert(10);  
        bt.insert(2);  
        bt.insert(5);  
        bt.insert(8);  
        bt.insert(17);  
        bt.insert(20); 
        bt.insert(15);
        bt.insert(25);
        
        //int target=5;
    //     if(bt.insert==target){
    //         System.out.println("Found");
    //     }
    //     else{
    //         System.out.println("Not Found");
    //     }
    // }
} 
}
